documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,462 @@
+
+
+.Producing messages with authorized access
+
+The `team-a-client` configuration is used to produce messages to topic `my-topic`:
+
+[source,shell]
+----
+bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9092 --topic my-topic \
+  --producer.config=/tmp/team-a-client.properties
+First message
+----
+
+A `Not authorized to access topics: [my-topic]` error is returned when trying to push the first message.
+
+`team-a-client` has a `Dev Team A` role that gives it permission to perform any supported actions on topics that start with `a_`, but can only write to topics that start with `x_`.
+The topic named `my-topic` matches neither of those rules.
+
+The `team-a-client` configuration is then used to produce messages to topic `a_messages`.
+
+[source,shell]
+----
+bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9092 --topic a_messages \
+  --producer.config /tmp/team-a-client.properties
+First message
+Second message
+----
+
+The messages are pushed out successfully, and in the Kafka container log there is DEBUG level output saying `Authorization GRANTED`.
+
+Use CTRL-C to exit the CLI application.
+
+You can see the Kafka container log by running:
+
+[source,shell]
+kubectl logs my-cluster-kafka-0 -f
+
+.Consuming messages with authorized access
+
+The `team-a-client` configuration is used to consume messages from topic `a_messages`:
+
+[source,shell,subs=+quotes]
+----
+bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic a_messages \
+  --from-beginning --consumer.config /tmp/team-a-client.properties
+----
+
+An error is returned as the `Dev Team A` role for `team-a-client` only has access to consumer groups that have names starting with `a_`.
+The `team-a-client` configuration is then used to consume messages when specifying a custom consumer group with a name that starts with `a_`:
+
+[source,shell,subs=+quotes]
+----
+bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic a_messages \
+  --from-beginning --consumer.config /tmp/team-a-client.properties --group a_consumer_group_1
+----
+
+This time the consumer receives all the messages from the `a_messages` topic.
+
+
+.Administering Kafka with authorized access
+
+The `team-a-client` configuration is used in administrative operations.
+
+Listing topics returns the `a_messages` topic.
+
+[source,shell]
+----
+bin/kafka-topics.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config /tmp/team-a-client.properties --list
+----    
+
+Listing consumer groups returns the `a_consumer_group_1` consumer group.
+
+[source,shell]
+----
+bin/kafka-consumer-groups.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config /tmp/team-a-client.properties --list
+----
+
+Fetching the default cluster configuration fails cluster authorization, because the operation requires cluster level permissions that `team-a-client` does not have.
```

Comment:
```suggestion
Fetching the default cluster configuration fails cluster authorization, because the operation requires cluster level permissions that `team-a-client` does not have:
```